### ISSUE TITLE
support for official version

### ISSUE THREAD

**Issue by pangaea47** [blake502/balatro-mobile-maker#118]
i just wanted to know if it is compatible with the official mobile version that released 3 weeks ago

**Comment by bokub** [comment]
I just tried to only transfer my saves to the official mobile version, but it doesn't work. There are no errors but I'm still at the beginning in the mobile version

**Comment by YthanZhang** [comment]
Also doesn't work for me, transfter shows no error, but have no effect.

**Comment by uday-sudo** [comment]
This won't work with the the official android version, because it is not debuggable and thus, it's files cannot be accessed as this project does with the steam version.

Even though there is no reliable way to move files from other platforms to the official android app.

You can certainly get your save files from the official android, using adb:

```
adb backup -noapk com.playstack.balatro.android
```

This will produce a backup.ab file which you can extract and go to:
```apps\com.playstack.balatro.android\f```

You will see the save files used by the official android app, this will only work one way not the other and sadly also out of scope for this project.

**Comment by blake502** [comment]
I'm tempted to allow the APK as a _source_ for building an unofficial APK _with_ debug flags. But that's out-of-scope for now.

**Comment by Kuroeks** [comment]
> This won't work with the the official android version, because it is not debuggable and thus, it's files cannot be accessed as this project does with the steam version.
> 
> Even though there is no reliable way to move files from other platforms to the official android app.
> 
> You can certainly get your save files from the official android, using adb:
> 
> ```
> adb backup -noapk com.playstack.balatro.android
> ```
> 
> This will produce a backup.ab file which you can extract and go to: `apps\com.playstack.balatro.android\f`
> 
> You will see the save files used by the official android app, this will only work one way not the other and sadly also out of scope for this project.

I did that and you can just rename files to make them work on steam version (i had tinkered like for 5 minutes? Just to double check). I had my save with my current run too. I don't know if the opposite is possible, but i think you just do the rename process in reverse and it should work

**Comment by BlindedFuture** [comment]
This won't work in reverse because for the official balatro app (com.playstack.balatro.android) the filesystem is not writeable without root priviledges. It can only be read by the backup method hence we can only get the save file but not put it back in.

**Comment by Kuroeks** [comment]
> This won't work in reverse because for the official balatro app (com.playstack.balatro.android) the filesystem is not writeable without root priviledges. It can only be read by the backup method hence we can only get the save file but not put it back in.

Yeah but you can actually Just repack the savedata and do `adb restore <backup name>.ab`

I extracted my savedata like this, so unless the restore uses some sort of checksum, It should be feasible. Tomorrow i'll try tho, and if It works i'll give detailed steps

**Comment by bokub** [comment]
I don't understand why the app developer(s) decided to store the save data in a root-only folder instead of a user-accessible one 🤔 

Stardew Valley for example makes it very easy to import / export saves
